### DISCUSSION TITLE
Marker as popup/overlay that doesnt impact chart-size

### DISCUSSION THREAD

**Discussion by zoltish** [patrykandpatrick/vico#327] [Ideas]
As soon as I include a marker in my charts, theres an added top padding that I think looks weird. For example, if my application follows a 8.dp grid, this sudden ~21.dp inset just breaks everything and I imagine somewhere in the world, a kitten dies. 

I think a viable solution would be to instead render the marker as part of an overlay/popup. If other elements are close to the chart, this temporary overlap is *imo* much preferrable to the constant added size/padding.

**Comment by Gowsky** [comment]
Hello @zoltish, the [`Marker`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.marker/-marker/?query=interface%20Marker%20:%20ChartInsetter) interface declares a [function](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.chart.insets/-chart-insetter/#1738844487%2FFunctions%2F186187275) which allows its subclasses to apply [`Insets`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.chart.insets/-insets/) to a chart. I suppose that you’re using [`MarkerComponent`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.component.marker/-marker-component/?query=open%20class%20MarkerComponent(val%20label:%20TextComponent,%20val%20indicator:%20Component?,%20val%20guideline:%20LineComponent?)%20:%20Marker) which applies the custom `Insets`. You can subclass the `MarkerComponent` and modify the default behavior.

**Reply by zoltish** [reply]
Hi @Gowsky! I forgot to mention that I've tried doing that as well. I can get rid of the added padding this way, but then the marker isn't rendering properly (just background, no text). I think it happens due to the layout ironically not having enough space for it - hence my thought about pop-up 💪🏽

**Reply by Gowsky** [reply]
Yes, you’d also need to override the drawing function, because the marker is drawn on top of a chart’s top bound.

**Reply by zoltish** [reply]
Interesting! I didn't think about that, I'll check it out. Thanks for the quick responses!

**Reply by zoltish** [reply]
I finally had a chance to test this out! Ultimately I feel like my initial request still stands, if I were to tweak the MarkerComponent.draw functionality to ditch the padding - either Ill (1) have the marker drawn on top of the chart, which looks weird, or (2) drawn outside of it (cut off by surrounding content). All of this is resolved by including it in a popup instead, although I understand that its a hefty change to make - so for now Ill get used to the added top-padding. If theres any other approach you think could work (i.e. drawing it inside a popup myself); feel free to share!

**Reply by patrickmichalik** [reply]
We’ll be looking into adding the feature requested. Thanks for the suggestion! I’m moving this to the “Ideas” category.
